cpp/wrappers/WKTJsiArrayWrapper.h

```diff
@@ -440,45 +440,58 @@ class JsiArrayWrapper : public JsiHostObject,
    */
   jsi::Value getArrayProxy(jsi::Runtime &runtime,
                            std::shared_ptr<jsi::HostObject> hostObj) {
-    // return jsi::Object::createFromHostObject(runtime, hostObj);
-
     auto createArrayProxy =
         runtime.global().getProperty(runtime, WorkletArrayProxyName);
+
+    // Install factory for creating an array proxy
     if (createArrayProxy.isUndefined()) {
       // Install worklet proxy helper into runtime
       static std::string code =
-          "function (obj) {"
-          "return new Proxy(obj, {"
-          "    ownKeys: function (target) {"
-          "      return Reflect.ownKeys(target).concat(['length']);"
-          "    },"
-          "    getPrototypeOf: function () {"
-          "      return Reflect.getPrototypeOf([]);"
-          "    },"
-          "    getOwnPropertyDescriptor: function (_, prop) {"
-          "      return {"
-          "        configurable: true,"
-          "        writable: true,"
-          "        enumerable: prop !== 'length',"
-          "      };"
-          "    },"
-          "    set: function(target, prop, value) { return "
-          "Reflect.set(target,prop,value); },"
-          "    get: function(target, prop) { return Reflect.get(target, prop); "
-          "}"
-          "  })"
-          "}";
-
+          "function (target) {"
+          "        const dummy = [];"
+          "        return new Proxy(dummy, {"
+          "          ownKeys: function (_target) {"
+          "            return Reflect.ownKeys(target).concat(['length']);"
+          "          },"
+          "          getOwnPropertyDescriptor: function (_target, prop) {"
+          "            return {"
+          "              ...Reflect.getOwnPropertyDescriptor(target, prop),"
+          "              configurable: prop !== 'length',"
+          "              writable: true,"
+          "              enumerable: prop !== 'length',"
+          "            };"
+          "          },"
+          "          set: function (_target, prop, value, _receiver) {"
+          "            return Reflect.set(target, prop, value, target);"
+          "          },"
+          "          get: function (_target, prop, receiver) {"
+          "            if (prop === 'length') return "
+          "Object.keys(target).length;"
```

Comment:
Isn't this now changed behaviour? Doesn't `.length` now include all functions, like `.map()`, etc as well?